src/bin/casr-cli.rs

```diff
@@ -629,3 +660,215 @@ fn change_text_view(layout1: &mut LinearLayout, act: Action) -> Option<EventResu
+                hash.push_str(san_line.as_str());
+                hash.push_str(casr_gdb_desc.as_str());
+                hash.push_str(casr_gdb_line.as_str());
+
+                let san_cnt = cluster_classes.get(&san_desc).unwrap_or(&0) + 1;
+                cluster_classes.insert(san_desc, san_cnt);
+                if !casr_gdb_desc.is_empty() {
+                    let casr_gdb_cnt = cluster_classes.get(&casr_gdb_desc).unwrap_or(&0) + 1;
+                    cluster_classes.insert(casr_gdb_desc, casr_gdb_cnt);
+                }
+
+                let value = if let Some(res) = cluster_hash.get(&hash) {
+                    (res.0.clone(), res.1 + 1)
+                } else {
+                    (result, 1)
+                };
+                cluster_hash.insert(hash, value);
+            }
+        }
+        for info in cluster_hash.values() {
+            info!("{}: {}", "Crash".green(), info.0.last().unwrap());
+            info!("  {}", info.0[0]);
+            if info.0.len() == 3 {
+                info!("  {}", info.0[1]);
+            }
+            info!("  Similar crashes: {}", info.1);
+        }
+        let mut classes = String::new();
+        cluster_classes.iter().for_each(|(class, number)| {
+            classes.push_str(format!(" {}: {}", class, number).as_str());
+            casr_classes.insert(
+                class.clone(),
+                casr_classes.get(class).unwrap_or(&0) + number,
+            );
+        });
+        info!("Cluster summary ->{}", classes);
+    }
+    let mut classes = String::new();
+    casr_classes
+        .iter()
+        .for_each(|(class, number)| classes.push_str(format!(" {}: {}", class, number).as_str()));
+    if classes.is_empty() {
+        info!("{} -> {}", "SUMMARY".magenta(), "No crashes found".red());
+    } else {
+        info!("{} ->{}", "SUMMARY".magenta(), classes);
+    }
+    Ok(())
+}
+
+/// Function processes report and returns summary
+///
+/// # Arguments
+///
+/// * 'report' - path to report
+///
+/// * 'extension' - casrep extension
+///
+/// # Return value
+///
+/// String - summary
+///
+fn process_report(report: &str, extension: &str) -> Result<(String, String, String)> {
+    if let Ok(file) = fs::File::open(report) {
+        if let Ok(jreport) = serde_json::from_reader(BufReader::new(file)) {
+            let jreport: Value = jreport;
+            let desc = jreport["CrashSeverity"]["ShortDescription"]
+                .as_str()
+                .unwrap()
+                .to_string();
+            let severity = jreport["CrashSeverity"]["Type"]
+                .as_str()
+                .unwrap()
+                .to_string();
+            let crashline = if let Some(crashline) = jreport.get("CrashLine") {
+                let crashline_str = crashline.as_str().unwrap();
+                if !crashline_str.trim().is_empty() {
+                    crashline_str.to_string()
+                } else {
+                    String::new()
```

Comment:
you could put `report` here

Reply:
Then if it was not possible to get the crashline from reports, hashes will be different for all reports